plugins/connectors/src/flyteplugins/connectors/bigquery/task.py

```diff
@@ -68,6 +72,8 @@ def custom_config(self, sctx: SerializationContext) -> Optional[Dict[str, Any]]:
         }
         if self.plugin_config.QueryJobConfig is not None:
             config.update(self.plugin_config.QueryJobConfig.to_api_repr()["query"])
+        if self.google_application_credentials is not None:
+            config["google_application_credentials"] = f"{{ .secrets.{self.google_application_credentials} }}"
```

Comment:
we dont need templating, we could simply add the "google_app_credentials_secret_key=key" to the Metadata, and then in connector look at all secrets to find the value that matches the key